### ISSUE TITLE
Sentry cron always `In Progress` to `Timeout`

### ISSUE THREAD

**Issue by eldrinchit** [getsentry/sentry#54664]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Sentry version: `@sentry/node@v7.58.1`

Cron configuration: 

<img width="715" alt="image" src="https://github.com/getsentry/sentry/assets/5326598/e1eebbc4-d0ea-48c4-bcfa-2ac4f68466d0">

I put this before the actual cron logic:
```
const checkInId = Sentry.captureCheckIn({
     monitorSlug: SCRIPT_NAME_LOWERCASE,
     status: 'in_progress',
})
```

this at the end: 
```
Sentry.captureCheckIn({
        checkInId,
        monitorSlug: SCRIPT_NAME_LOWERCASE,
        status: 'ok',
      })
```

But it will always get timed out, even though the script was successfully ran.

### Expected Result

Successful status for cron checkin

### Actual Result

<img width="845" alt="image" src="https://github.com/getsentry/sentry/assets/5326598/83d54eb8-b108-4936-977d-8a4effc14975">

### Product Area

Crons

### DSN

https://f88e4f8c9cc64480be1cbae1fe7a2fcf@o1193026.ingest.sentry.io/6324830

**Comment by evanpurkhiser** [comment]
Hmm, yeah you definitely should be seeing successful checkins.

Can you confirm in your debug logs that when the `ok` check-in is sent, the check-in ID is the same as the `in_progress` check-in is sent.

I see you mentioned

> Also checkinID, always has a value

But just wanted to verify!

**Comment by AbhiPrasad** [comment]
Wasn't able to reproduce this problem.

Could you add the following code to your app and see what check in ids are being logged?

```js
Sentry.getCurrentHub()
  .getClient()
  .on("beforeEnvelope", (envelope) => {
    const item = envelope[1][0];
    if (item[0].type === "check_in") {
      console.log(item);
    }
  });
```

**Comment by xenon92** [comment]
Well, I came to this thread as I was facing the same issue with Nodejs SDK.

Using the suggested listener above to verify the check in ids being generated in both the function calls, I could see that they were different.

I was using a different variable name '**sentryCheckInId**' instead of 'checkInId' as mentioned in the documentation to capture the id from **in_progress** call and then passing it to the '**ok**' call.

> 
> ```js
> Sentry.getCurrentHub()
>   .getClient()
>   .on("beforeEnvelope", (envelope) => {
>     const item = envelope[1][0];
>     if (item[0].type === "check_in") {
>       console.log(item);
>     }
>   });
> ```

Solution: For the OK function call, I passed the sentryCheckInId using the parameter name 'checkInId' found in the interface FinishedCheckIn in the SDK and the issue was resolved.

`
Sentry.captureCheckIn({
        checkInId: sentryCheckInId,
        monitorSlug: SCRIPT_NAME_LOWERCASE,
        status: 'ok',
      })
`

I would suggest to update the documentation and mention the parameter name **checkInId** in the object being passed in the 'ok' call as you've done for **monitorSlug** and **status** parameter names. This ensures your documentation will always work for the end user even if he chooses a different variable name than 'checkInId'.

**Comment by AbhiPrasad** [comment]
Hey @xenon92 - thanks for your feedback! I don't think we're going to change the docs because a) using TypeScript would have caught this issue because of our typing and b) the shorthand notation is pretty common.

I do recognize this is still a pain point, so we're going to be introducing a new API with https://github.com/getsentry/sentry-javascript/pull/9395 that abstracts away setting the ids. Feedback appreciated in that PR!

```js
import * as Sentry from '@sentry/node';

const result = Sentry.withMonitor(
  'dailyEmail',
  () => {
    // withCheckIn return value is same return value here
    return sendEmail();
  },
  {
    schedule: {
      type: 'crontab',
      value: '0 * * * *',
    },
    // 🇨🇦🫡
    timezone: 'Canada/Eastern',
  },
);
```

**Comment by jeengbe** [comment]
I have to +1, it's weird that checking out still requires both the monitor name plus the check-in ID. Feels like the latter should be enough. I guess the issue lies in the API, though.

**Comment by AbhiPrasad** [comment]
I just shipped new API https://github.com/getsentry/sentry-javascript/releases/tag/7.76.0, writing docs for it here: https://github.com/getsentry/sentry-docs/pull/8347

Yeah there's some backend reasons why checkInId is needed, but we should be able to abstract that away from users.

**Comment by jeengbe** [comment]
Yeah it's the backend API I meant 🙂 but it's weird that it requires the monitor name. 

This call:

```bash
  # Error
  curl -X PUT \
      "https://sentry.io/api/0/organizations/ORG/monitors/$SENTRY_MONITOR_ID/checkins/$check_in_id/" \
      --header "Authorization: DSN $SENTRY_DSN" \
      --header 'Content-Type: application/json' \
      --data-raw '{"status": "error"}'
```

really shouldn't require the monitor ID. I can see why it's there but that makes it more cumbersome to use

**Comment by AbhiPrasad** [comment]
A project (which is what a DSN maps to) can have multiple distinct monitors, so we need some kind of unique identifier to direct the event to the correct place when processing/storing the check in.

check in id is simply used to related `in_progress` check in to `error`/`ok` check in.
